src/Psy/Plugin/AbstractPlugin.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Psy\Plugin;
+
+abstract class AbstractPlugin
+{
+    public static function register()
+    {
+        Manager::register(new static(), static::getName());
+    }
+
+    /**
+     * @return string
+     */
+    final public static function getName()
+    {
+        $class = new \ReflectionClass(get_called_class());
+
+        return preg_replace('#Plugin$#', '', $class->getShortName());
+    }
+
+    /**
+     * @param array $configuration
+     *
+     * @return array
+     */
+    final public static function getConfiguration($configuration = array())
```

Comment:
I've been thinking about this method, and I do agree that having "2" entry point (one through a `getConfiguration` and one through the getters) is two much. More over, this method is doing a merge to "chain" the methods, and I don't think that this should be the scope of this method. it should just return a plain array if it is left as-is